modules/tinymce/src/themes/silver/test/ts/browser/view/ViewButtonsContextTest.ts

```diff
@@ -0,0 +1,214 @@
+    clickViewButton(editor, 'button-with-toggle');
+    assert.notEqual(getSvg(editor, 'button-with-toggle'), initialButtonWithToggleButtonSvg, 'after the first toggle icon should change');
+    clickViewButton(editor, 'button-with-toggle');
+    assert.equal(getSvg(editor, 'button-with-toggle'), initialButtonWithToggleButtonSvg, 'after the second toggle icon should return to the old value');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'button-with-toggle' ]);
+
+    assertButtonNativelyEnabled('button-without-toggle');
+    const initialbuttonWithoutToggleButtonSvg = getSvg(editor, 'button-without-toggle');
+    clickViewButton(editor, 'button-without-toggle');
+    assert.equal(getSvg(editor, 'button-without-toggle'), initialbuttonWithoutToggleButtonSvg, 'click should not toggle icon');
+    clickViewButton(editor, 'button-without-toggle');
+    assert.equal(getSvg(editor, 'button-without-toggle'), initialbuttonWithoutToggleButtonSvg, 'click should not toggle icon');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'button-with-toggle', 'button-without-toggle' ]);
+  });
+
+  it('TINY-9616: if is active is true the button should have ViewButtonClasses.Ticked', async () => {
+    const editor = hook.editor();
+    toggleView('myview1');
+    await UiFinder.pWaitFor('buttons should be showed', TinyDom.container(editor), '[aria-label="button-active-true"]');
+
+    const buttonActiveTrue = getButtonByTitle('button-active-true');
+    assert.isTrue(Class.has(buttonActiveTrue, ViewButtonClasses.Ticked), 'button with active true should have ticked class');
+    const buttonActiveFalse = getButtonByTitle('button-active-false');
+    assert.isFalse(Class.has(buttonActiveFalse, ViewButtonClasses.Ticked), 'button with active false should not have ticked class');
+    const buttonNoActive = getButtonByTitle('button-no-active');
+    assert.isFalse(Class.has(buttonNoActive, ViewButtonClasses.Ticked), 'button without active flag should not have ticked class');
+
+    clickViewButton(editor, 'button-active-true');
+    clickViewButton(editor, 'button-active-false');
+    clickViewButton(editor, 'button-no-active');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'button-active-true', 'button-active-false', 'button-no-active' ]);
+  });
+
+  it('TINY-9616: if is active is t2 rue the button should have ViewButtonClasses.Ticked', async () => {
+    const editor = hook.editor();
+    editor.mode.set('testmode');
+    toggleView('myview1');
+    await UiFinder.pWaitFor('buttons should be showed', TinyDom.container(editor), '[aria-label="Save code"]');
+
+    clickViewButton(editor, 'Save code');
+    clickViewButton(editor, 'Cancel');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'Save code', 'Cancel' ]);
+  });
+
+  it('TINY-9616: View button without context defaults to mode:design', async () => {
```

Comment:
Chang the ID on this one and maybe the title that clicks are allowed or something.